libs/designsystem/header/src/header.component.spec.ts

```diff
@@ -0,0 +1,168 @@
+    </kirby-header>
+    `;
+
+    let spectator: Spectator<HeaderComponent>;
+    beforeEach(() => {
+      spectator = createHost(defaultTemplate);
+    });
+
+    it(`should have correct title`, () => {
+      const titleElement = getTitleElement(spectator);
+
+      expect(titleElement).toContainText(title);
+      expect(titleElement).toHaveComputedStyle({
+        'font-size': '32px',
+      });
+    });
+
+    it(`should have correct subtitle1`, () => {
+      const subtitle1Element = getSubtitle1Element(spectator);
+
+      expect(subtitle1Element).toContainText(subtitle1);
+      expect(subtitle1Element).toHaveComputedStyle({
+        'font-size': '14px',
+      });
+    });
+
+    it(`should have subtitle2`, () => {
+      const subtitle2Element = getSubtitle2Element(spectator);
+
+      expect(subtitle2Element).toContainText(subtitle2);
+      expect(subtitle2Element).toHaveComputedStyle({
+        'font-size': '14px',
+      });
+    });
+  });
+
+  describe('Value', () => {
+    const title = 'title';
+    const value = '12.345,67';
+    const valueUnit = 'USD';
+
+    const defaultTemplate = `
+    <kirby-header title="${title}" value="${value}" valueUnit="${valueUnit}">
+    </kirby-header>
+    `;
+
+    let spectator: Spectator<HeaderComponent>;
+    beforeEach(() => {
+      spectator = createHost(defaultTemplate);
+    });
+
+    it(`should have correct title`, () => {
+      const keyElement = getKeyElement(spectator);
+
+      expect(keyElement).toContainText(title);
+      expect(keyElement).toHaveComputedStyle({
+        'font-size': '18px',
+      });
+    });
+    it(`should have correct value`, () => {
+      const valueElement = getValueElement(spectator);
+
+      expect(valueElement).toContainText(value);
+      expect(valueElement).toHaveComputedStyle({
+        'font-size': '40px',
+      });
+    });
+
+    it(`should have correct valueUnit`, () => {
+      const valueUnitElement = getValueUnitElement(spectator);
+
+      expect(valueUnitElement).toContainText(valueUnit);
+      expect(valueUnitElement).toHaveComputedStyle({
+        'font-size': '16px',
+      });
+    });
+  });
+
+  describe('Avatar', () => {
+    const title = 'title';
+    const subtitle1 = 'Subtitle one';
+    const subtitle2 = 'Subtitle two';
+
+    const defaultTemplate = `
+    <kirby-header title="${title}" subtitle1="${subtitle1}" subtitle2="${subtitle2}">
+      <kirby-avatar size="lg" text="A" title="lg"></kirby-avatar>
+    </kirby-header>
+    `;
+
+    let spectator: Spectator<HeaderComponent>;
+    beforeEach(() => {
+      spectator = createHost(defaultTemplate);
+    });
+
+    it(`should have correct avatar`, () => {
+      const avatarElement = getAvatarElement(spectator);
+
+      expect(avatarElement).toBeTruthy();
+    });
+  });
+
+  describe('Flag', () => {
+    const title = 'title';
+    const subtitle1 = 'Subtitle one';
+    const subtitle2 = 'Subtitle two';
+
+    const defaultTemplate = `
```

Comment:
Consider inlining the template, as it's only used once (and aligns rest of test code)